ticoniq/jobconiq
# Feature/developer dashboard

# Pull Request Description

## Feature: Developer Dashboard

### Overview

This pull request introduces the Developer Dashboard feature. The Developer Dashboard allows developers to view and manage their profiles, applications, and company information in a centralized location.

### Changes

#### 1. Developer Dashboard Page
- Added a new `Developer Dashboard` component that serves as the main page for developers.
- The dashboard provides an overview of the developer's profile, recent activities, and applications.

#### 2. Applied job List Display
- Displayed a list of all job applied by a single user.
- Added a section in the Developer Dashboard to display all companies that the developer has interacted with.

#### 2. Increment Job View Count
- Implemented the `incrementJobViewCount` function to track and update the view count for job listings.
- Fixed issues related to the `jobViewCount` model not being defined correctly in Prisma schema.
- Ensured the job view count updates and revalidates the job page to reflect changes in real-time.

### Bug Fixes
- Fixed an issue where the `jobViewCount` model was not being accessed correctly, causing runtime errors.
- Ensured proper error handling and logging for the `incrementJobViewCount` function.

### Code Changes
- `src/app/(frontend)/jobs/incrementJobViewCount.ts`
- `src/app/(user)/developer/dashboard.tsx`
- `src/components/ui/button.tsx`
- `src/components/ui/card.tsx`
- `src/components/ui/input.tsx`

### Additional Improvements
- Improved UI/UX for the Developer Dashboard, making it more intuitive and user-friendly.
- Enhanced error handling and logging for better debugging and maintenance.

### Testing
- Thoroughly tested the new Developer Dashboard feature to ensure all functionalities work as expected.
- Verified the job view count increments and updates correctly.

### Screenshots
![Screenshot_10-7-2024_175242_www jobconiq live](https://github.com/ticoniq/jobconiq/assets/71973088/b767e906-9bea-4fb9-aec5-97c55995d035)

### How to Test
1. Checkout the `Feature/developer-dashboard` branch.
2. Navigate to the developer dashboard page.
3. Verify the list of companies is displayed on the main page and in the developer dashboard.
4. Click on a company to view its details.
5. Check the job listings and ensure the view count increments correctly when viewed.

### Notes
- Ensure your Prisma schema is updated to include the `jobViewCount` model as specified in the code changes.

### Future Improvements
- Implement additional filtering and sorting options for the company list.
- Add more detailed analytics and insights on the Developer Dashboard.

### Closing
This feature enhances the developer experience by providing a centralized dashboard for managing profiles, applications, and company interactions. It also improves job listing analytics through the view count tracking functionality.

---

Please review the changes and provide feedback. Thank you!
